src/GraphManager/GraphEdit/ButtonTests.test.tsx

```diff
@@ -0,0 +1,149 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { NoTouchButton } from "./NoTouchButton";
+import { UserSettingsButton } from "./UserSettingsButton";
+import { EditModeButton } from "./ModeButton";
+import { CreateButton} from "./CreateButton";
+import { Controller } from "./GraphEdit";
+
+// Mock functions from GraphEdit
+jest.mock("./GraphEdit", () => ({
+  openCreateNodePopUpAtPagePosition: jest.fn(),
+  openCreateLinkPopUp: jest.fn(),
+  i18n: { t: jest.fn((key: string) => key) }, // Mock translation function
+}));
+
+// Mock User Data Context
+jest.mock("@src/Context/UserDataContext", () => ({
+  useUserDataContext: jest.fn(),
+}));
+
+import { useUserDataContext } from "@src/Context/UserDataContext";
+
+describe("Button Components", () => {
+  let mockController: Controller;
+  let setIsOpen: jest.Mock;
+  let mockDisplayAlert: jest.Mock;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    // Configure the mocked useUserDataContext
+    (useUserDataContext as jest.Mock).mockReturnValue({ userID: null });
+
+    // Mock necessary parts of the Controller
+    mockController = {
```

Comment:
we have a function called `makeMockController()` that should be used here instead of creating a partial mock manually inline